Game2048Intelligence/src/main/java/heuristic/PreferUpHeuristic.java

```diff
@@ -0,0 +1,47 @@
+package heuristic;
+
+import models.game2048.KeyEventHandler;
+import models.game2048.Scoreboard;
+import models.grid2048.Grid2048;
+import java.util.Random;
+
+public class PreferUpHeuristic extends GameHeuristic {
+    public PreferUpHeuristic(Grid2048 grid2048, Scoreboard scoreboard) {
+        super(grid2048, scoreboard);
+    }
+
+    @Override
+    public KeyEventHandler evaluateNextGameState() {
+        KeyEventHandler[] eventSequence = establishKeyEventSequence();
+        int idx = 0;
```

Comment:
This code could be made much less confusing by using the simple enum once the logic has been moved out of it. Rather than iterating over an obtuse index, we could have establishKeyEventSequence return a List<Grid2048Event> (I actually would rename that, "event" seems nondescript. Maybe Grid2048Input). Then we could easily iterate over the list of events and try them each in turn.

Reply:
Check my new implementation.